### PR TITLE
Add binary generation for releases

### PR DESCRIPTION
**Pull request by saswatamcode**
This PR adds support for binary generation for different platforms during release.
It uses `promu` and a CircleCI job to automate this (the config has been mostly copied from Thanos). Would need some maintainer help in setting this up in CircleCI. 🙂 

This addresses the discussion in #38. This project will still be installable via `go install`.

I've used `promu` here as that is what I'm familiar with and is also used by larger projects like Thanos, Prometheus, Alertmanager etc. Feel free to recommend alternatives that might be better!

### CODE REVIEW

**Comment by douglascamata** [comment]
I love to use [GoReleaser](https://goreleaser.com) for these cases. It's easy to use and does the job very well.

**Comment by matej-g** [comment]
> I love to use [GoReleaser](https://goreleaser.com) for these cases. It's easy to use and does the job very well.

Hm yes, I think there were also some discussion in Thanos about switching to GoReleaser, but as of now it looks like most projects in our ecosystem use `promu` as @saswatamcode said in description and that's probably what we have most experience with.

So I'm not opposed to using `promu` for this initial work and we could evaluate GoReleaser sometime in the future again. WDYT @saswatamcode @douglascamata?

**Comment by saswatamcode** [comment]
Sure, I think GoReleaser might be a great alternative that we can/might need to adopt in the future (even for other projects)!. 
Let's try `promu` for now, and see if we face difficulties! It might be good to contribute and fix things upstream too so that projects using `promu` in the ecosystem can benefit from it!  🙂

File: docs/release.md [hunk 1]
```diff
@@ -2,13 +2,14 @@
 
 We use semantic versioning for `obsctl` in format `vX.Y.Z`.
 
-Currently, we do not include binaries with our releases. Instead, users are instructed to install with `go install`.
+We have started including binaries after initial release. Users can also use `go install` to install obsctl.
```
Comment (moadz) [inline docs/release.md:5]:
```suggestion
We also ship binaries <link to readme>. Users can also use `go install` to install obsctl.
```

Passive voice, we should probably just link to instructions on how to get the binary.

**Review by matej-g — APPROVED** [review]
Looks good, nice work! Let's have a final agreement on `promu` vs GoReleaser, then I'm happy to let this change in :slightly_smiling_face:

File: docs/release.md (hunk 1 above)
Comment (saswatamcode) [reply docs/release.md]: Yup, changed, I'll update the readme install section once this change is in, and we have binaries.

**Comment by douglascamata** [comment]
I prefer using GoReleaser for being accessible and simple to use. There is a small subset of Go devs out there (mostly people involved in project in the Prometheus ecosystem) that know about `promu` and how to use it, while there are plenty that have had some experience with GoReleaser. I noticed that `promu` has very little documentation that I would find, while GoReleaser has plenty.

**Comment by douglascamata** [comment]
Please don't consider my comment a blocker though. 😄

**Comment by saswatamcode** [comment]
Sure! I think we kind of fall within the small subset of Go + Prometheus ecosystem, especially with Observatorium being an adjacent org to Thanos. I think there is a push within ecosystem to move to GoReleaser too e.g https://github.com/thanos-io/thanos/issues/5343. So I think it might be worth moving to GoReleaser when more mature projects do, as that way we'll have potential things to reuse like configs or even things like CircleCI orbs. 🙂

**Pull request merged**